lib/pubid/core/identifier/base.rb

```diff
@@ -167,6 +167,19 @@ def resolve_stage(stage)
         # @typed_stage = self.class::TYPED_STAGES[@typed_stage][:abbr] if @typed_stage
       end
 
+      # Checks if another identifier is newer edition of the same document
+      # @param other [Pubid::Core::Identifier::Base] pubid identifier to compare with
+      # @return [Boolean] true if another identifier is newer edition
+      def new_edition_of?(other)
+        raise Errors::AnotherDocumentError, "cannot compare edition with #{other}" unless exclude(:year) == other.exclude(:year)
+
+        return false if year.nil?
+
+        return true if other.year.nil?
+
+        year < other.year
```

Comment:
You need to also check the case where if they have edition numbers, some may not have year but have edition.

Reply:
> You need to also check the case where if they have edition numbers, some may not have year but have edition.

@ronaldtse I have question here about that: https://github.com/metanorma/pubid-iso/issues/274
What if we have identifier year 1999, edition number 2 and another identifier year 2000 and edition number 1, is it possible? Does edition number depends on year or they are independed? Is it the same for all publishers? (ISO, IEEE, NIST, ITU, etc)

Reply:
It is possible, what that means is that the new edition 1 is a different document altogether from the old document. 

This happens when a standard document (eg 10001 edition 3) is re-numbered to another number (eg 11001 edition 3), then a new document adopts the old number (eg 10001 edition 1). 

So this means these two documents are "incomparable". As @andrew2net mentioned, we need to distinguish cases where the documents cannot be compared. Either use NIL or throw an error.